File: src/algorithmic_toolbox/week2/LCM.java
package algorithmic_toolbox.week2;

import java.util.Scanner;


public class LCM {

  public static int getGcd(int a, int b) {
    if (b == 0) {
      return a;
    }
    return getGcd(b, a % b);
  }

  public static long getLcm(int a, int b) {
    long hcf = (a > b) ? getGcd(a, b) : getGcd(b, a);

    return (a / hcf) * (b / hcf);
  }

  public static void main(String[] args) {
    Scanner scanner = new Scanner(System.in);
    int a = scanner.nextInt();
    int b = scanner.nextInt();
    scanner.close();

    System.out.println(getLcm(a, b));
  }
}

File: src/algorithmic_toolbox/week2/lcm.py
def gcd(a: int , b: int) -> int:
    if b == 0:
        return a
    return gcd(b, a % b)


def lcm(a: int, b: int) -> int:
    hcf = gcd(a, b) if a > b else gcd(b, a)
    return (a * b) // hcf


if __name__ == '__main__':
    a, b = [int(x) for x in input().split()]
    print(lcm(a, b))
